neurom/core/soma.py

```diff
@@ -203,13 +203,24 @@ def __str__(self):
         )
 
 
-def soma_center(soma):
-    """Calculate soma center."""
+def get_morphio_soma(obj):
+    """Get the MorphIO soma from the given object."""
+    if hasattr(obj, "soma"):
+        input_soma = obj.soma
+    else:
+        input_soma = obj
 
-    if hasattr(soma, "to_morphio"):
-        morphio_soma = soma.to_morphio()
+    if hasattr(input_soma, "to_morphio"):
+        morphio_soma = input_soma.to_morphio()
     else:
-        morphio_soma = soma
+        morphio_soma = input_soma
+
+    return morphio_soma
```

Comment:
This is not needed. The functions work with both somas. Passing a morphology here is out of the scope of this module.

Reply:
I still think it makes sense to write something like `soma_volume(morph)`, don't you? We can move the functions to another module and keep only the Soma class here if that bothers you.